homeassistant/components/tplink/__init__.py

```diff
@@ -126,3 +186,29 @@ def legacy_device_id(device: SmartDevice) -> str:
     if "_" not in device_id:
         return device_id
     return device_id.split("_")[1]
+
+
+def _get_store(hass: HomeAssistant) -> Store[dict[str, dict[str, str]]]:
+    return Store[dict[str, dict[str, str]]](hass, STORAGE_VERSION, STORAGE_KEY)
+
+
+async def get_stored_credentials(hass: HomeAssistant) -> Optional[Credentials]:
+    """Retrieve the credentials from the Store."""
+    storage_data = await _get_store(hass).async_load()
+    if storage_data and (auth := storage_data[CONF_AUTHENTICATION]):
+        return Credentials(auth[CONF_USERNAME], auth[CONF_PASSWORD])
+
+    return None
+
+
+async def set_stored_credentials(
+    hass: HomeAssistant, username: str, password: str
+) -> None:
+    """Save the credentials to the Store."""
+    storage_data = {
+        CONF_AUTHENTICATION: {
+            CONF_USERNAME: username,
+            CONF_PASSWORD: password,
+        }
+    }
+    await _get_store(hass).async_save(storage_data)
```

Comment:
If we have a single storage we can use the schedule save function instead so it can group writes to minimize I/O

Reply:
There will only be a write the first time the user enters credentials, after that only reads.  Is that still a concern?

Reply:
Store is now dropped